### PR TITLE
[SPARK-44134][CORE] Fix setting resources (GPU/FPGA) to 0 when they are set in spark-defaults.conf

### PR DESCRIPTION
**Pull request by tgravescs**
### What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/SPARK-44134

With resource aware scheduling, if you specify a default value in the spark-defaults.conf, a user can't override that to set it to 0.

Meaning spark-defaults.conf has something like:
spark.executor.resource.{resourceName}.amount=1
spark.task.resource.{resourceName}.amount =1

If the user tries to override when submitting an application with spark.executor.resource.{resourceName}.amount=0 and spark.task.resource.{resourceName}.amount =0, the applicatoin fails to submit.  it should submit and just not try to allocate those resources. This worked back in Spark 3.0 but was broken when the stage level scheduling feature was added.

Here I fixed it by simply removing any task resources from the list if they are set to 0.

Note I also fixed a typo in the exception message when no executor resources are specified but task resources are.

Note, ideally this is backported to all of the maintenance releases

### Why are the changes needed?

Fix a bug described above

### Does this PR introduce _any_ user-facing change?

no api changes

### How was this patch tested?

Added unit test and then ran manually on standalone and YARN clusters to verify overriding the configs now works.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala [hunk 1]
```diff
@@ -215,7 +215,7 @@ class ResourceProfile(
       }
     }
     if (taskResourcesToCheck.nonEmpty) {
-      throw new SparkException("No executor resource configs were not specified for the " +
+      throw new SparkException("No executor resource configs were specified for the " +
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala:218]: Nice catch!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

cc @mridulm too

**Comment by dongjoon-hyun** [comment]
This is not a release blocker because the affected version is 3.2.0.

BTW, if you don't mind, could you participate Apache Spark 3.4.1 RC1 vote, @tgravescs ?

**Comment by tgravescs** [comment]
thanks Dongjoon.  Sure, I'll look at the rc1.

**Comment by dongjoon-hyun** [comment]
Thank you, @tgravescs !

**Review by HyukjinKwon — COMMENTED** [review]
LGTM2

**Pull request closed without merging**